### PR TITLE
core(font-display): more accurately follow CSS spec

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
One of the reasons people seem to be getting false failures is our semicolon. Thanks to @Diklla for pointing this out!

**Related Issues/PRs**
#6628

### CODE REVIEW

File: lighthouse-core/audits/font-display.js [hunk 1]
```diff
@@ -57,7 +57,9 @@ class FontDisplay extends Audit {
       const fontFaceDeclarations = newlinesStripped.match(/@font-face\s*{(.*?)}/g) || [];
       // Go through all the @font-face declarations to find a declared `font-display: ` property
       for (const declaration of fontFaceDeclarations) {
-        const rawFontDisplay = declaration.match(/font-display:(.*?);/);
+        // Find the font-display value by matching a single token, optionally surrounded by whitespace,
+        // followed either by a semicolon or the end of a block.
+        const rawFontDisplay = declaration.match(/font-display:(\s*\S+\s*)(;|\s*\})/);
```
Comment (brendankenny) [inline lighthouse-core/audits/font-display.js]: is the last `\s*` not already captured by the first group?

Comment (brendankenny) [reply lighthouse-core/audits/font-display.js]: after just getting burned by the old regex, I feel like we need some extra whitespace tests for the cases we're going to be handling now :)

**Comment by wardpeet** [comment]
@patrickhulce do you mind adding a test where you check all possibilities from the regex?
```
display:  'optional'    ;
```
```
display:  'optional' }
```
```
display:  'optional'}
```

File: lighthouse-core/audits/font-display.js (hunk 1 above)
Comment (exterkamp) [reply lighthouse-core/audits/font-display.js]: +1 to @brendankenny first comment, should be:
```
font-display:(\s*\S+\s*)(;|\})
```
iirc it is valid to have white space after the property before the colon like:
```css
font-display : 'optional';
```
in which case we would need:
```
font-display\s*:(\s*\S+\s*)(;|\})
```

**Comment by patrickhulce** [comment]
great points everyone!

since at this point we're just taking a step back on the whole audit, I thought I'd tighten it up and make it more correct rather than the loosely goosey forgiving audit that it was before :)

File: lighthouse-core/audits/font-display.js [hunk 2]
```diff
@@ -57,11 +57,13 @@ class FontDisplay extends Audit {
...
       for (const declaration of fontFaceDeclarations) {
-        const rawFontDisplay = declaration.match(/font-display:(.*?);/);
+        // Find the font-display value by matching a single token, optionally surrounded by whitespace,
+        // followed either by a semicolon or the end of a block.
+        const rawFontDisplay = declaration.match(/font-display\s*:(\s*\w+\s*)(;|\})/);
         // If they didn't have a font-display property, it's the default, and it's failing; bail
         if (!rawFontDisplay) continue;
         // If they don't have one of the passing font-display values, it's failing; bail
-        const hasPassingFontDisplay = PASSING_FONT_DISPLAY_REGEX.test(rawFontDisplay[0]);
+        const hasPassingFontDisplay = PASSING_FONT_DISPLAY_REGEX.test(rawFontDisplay[1].trim());
```
Comment (brendankenny) [inline lighthouse-core/audits/font-display.js]: nit: worth keeping the whitespace in the capture group if only going to trim it?

File: lighthouse-core/test/audits/font-display-test.js [hunk 3]
```diff
@@ -72,32 +74,32 @@ describe('Performance: Font Display audit', () => {
...
-    assert.deepEqual(result.details.items, items);
+    expect(result.details.items).toEqual(items);
   });
 
   it('resolves URLs relative to stylesheet URL when available', async () => {
     stylesheet.header.sourceURL = 'https://cdn.example.com/foo/bar/file.css';
     stylesheet.content = `
       @font-face {
-        font-display: 'block';
+        font-display: block;
```
Comment (brendankenny) [inline lighthouse-core/test/audits/font-display-test.js]: ha, whoops

**Comment by schoenwaldnils** [comment]
In which lighthouse version is this published?
I still get the same failing audit on this site: https://rw.schoen.world/ :/

Report: https://lighthouse-dot-webdotdevsite.appspot.com/lh/html?url=https://rw.schoen.world/

Minified styles in the head using emotion.js
```
@font-face{font-family:'Open Sans';font-style:normal;font-weight:300;font-display:swap;src:local('Open Sans Light'),local('OpenSans-Light'),url(https://fonts.gstatic.com/s/opensans/v16/mem5YaGs126MiZpBA-UN_r8OUuhpKKSTjw.woff2) format('woff2')}@font-face{font-family:'Open Sans';font-style:normal;font-weight:400;font-display:swap;src:local('Open Sans Regular'),local('OpenSans-Regular'),url(https://fonts.gstatic.com/s/opensans/v16/mem8YaGs126MiZpBA-UFVZ0bf8pkAg.woff2) format('woff2')}@font-face{font-family:'Open Sans';font-style:normal;font-weight:600;font-display:swap;src:local('Open Sans SemiBold'),local('OpenSans-SemiBold'),url(https://fonts.gstatic.com/s/opensans/v16/mem5YaGs126MiZpBA-UNirkOUuhpKKSTjw.woff2) format('woff2')}
```

**Comment by paulirish** [comment]
This fix is shipping in v4.2.0+

**Pull request merged**